include/ifc/pathname.hxx

```diff
@@ -71,7 +72,9 @@ namespace ifc {
 
         std::size_t length() const
         {
-            return Base::size() - 1;
+            std::size_t size = Base::size();
+            IFCASSERT(size != 0);
```

Comment:
Are these really best handled with asserts?

Reply:
No, these asserts are overkill and unnecessary.  The constructors always ensure that `Base::size()` is non-zero.

Reply:
Ah, right. They always push back a default value, so the size is never zero.